### ISSUE TITLE
Change batch requests to run concurrently

### ISSUE THREAD

**Issue by oeed** [async-graphql/async-graphql#1290]
[Currently](https://github.com/async-graphql/async-graphql/blob/382d54d5817e1bd3259c01712c76f6edc50d1081/src/schema.rs#L519) batch requests appear to run each request one by one, rather than at the same time. This loses some of the value of async code, as each request in the batch ends up running synchronously.

```rust
futures_util::stream::iter(requests.into_iter())
    .then(|request| self.execute(request))
    .collect()
    .await
```

From testing this in dummy project, this way of iterating is done sequentially. Is there a fundamental reason these requests can't be run concurrently? I'm open to doing a PR if not.

Some of my queries can take multiple seconds, so when sent in a batch the total request can take significantly longer than if they were run concurrently.

---

To illustrate, the following example:
```rust
#[tokio::main]
async fn main() {
    let req = vec![1, 2, 3, 4, 5];
    let _: Vec<_> = futures_util::stream::iter(req.into_iter())
        .then(|i| async move {
            println!("Start {}", i);
            tokio::time::sleep(std::time::Duration::from_secs(2)).await;
            println!("End {}", i);
        })
        .collect()
        .await;
}
```

Results in this console output, notice they happen sequentially:
```
Start 1
End 1
Start 2
End 2
Start 3
End 3
Start 4
End 4
Start 5
End 5
```

**Comment by sunli829** [comment]
This is a bug and I'm working on it. 🙂

**Comment by sunli829** [comment]
fixed in `v0.5.9`

**Comment by frederikhors** [comment]
Can we close this?
